lxd/container.go

```diff
@@ -748,22 +748,24 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 
 func containerConfigureInternal(c container) error {
 	// Find the root device
-	for _, m := range c.ExpandedDevices() {
-		if m["type"] != "disk" || m["path"] != "/" || m["size"] == "" {
-			continue
-		}
+	localDevices := c.LocalDevices()
+	_, rootDiskDevice := containerGetRootDiskDevice(localDevices)
+	if rootDiskDevice["pool"] == "" {
+		expandedDevices := c.ExpandedDevices()
+		_, rootDiskDevice = containerGetRootDiskDevice(expandedDevices)
+	}
```

Comment:
ExpandedDevices applies each profile in sequence, overriding entire device entries (not device properties) when another one of the same name is found. It then applies LocalDevices on top of that, overriding anything that came from a profile if the container has it in its own devices.

That's how the whole inheritance works for Devices and Config throughout LXD.